src/main/java/org/mockito/internal/creation/bytebuddy/MockMethodAdvice.java

```diff
@@ -344,6 +366,122 @@ boolean checkSelfCall(Object value) {
         }
     }
 
+    static class ConstructorShortcut
+            implements AsmVisitorWrapper.ForDeclaredMethods.MethodVisitorWrapper {
+
+        private final String identifier;
+
+        ConstructorShortcut(String identifier) {
+            this.identifier = identifier;
+        }
+
+        @Override
+        public MethodVisitor wrap(
+                TypeDescription instrumentedType,
+                MethodDescription instrumentedMethod,
+                MethodVisitor methodVisitor,
+                Implementation.Context implementationContext,
+                TypePool typePool,
+                int writerFlags,
+                int readerFlags) {
+            if (instrumentedMethod.isConstructor() && !instrumentedType.represents(Object.class)) {
+                MethodList<MethodDescription.InDefinedShape> constructors =
+                        instrumentedType
+                                .getSuperClass()
+                                .asErasure()
+                                .getDeclaredMethods()
+                                .filter(isConstructor().and(not(isPrivate())));
+                int arguments = Integer.MAX_VALUE;
+                boolean visible = false;
+                MethodDescription.InDefinedShape current = null;
+                for (MethodDescription.InDefinedShape constructor : constructors) {
+                    if (constructor.getParameters().size() < arguments
+                            && (!visible || constructor.isPackagePrivate())) {
+                        current = constructor;
+                        visible = constructor.isPackagePrivate();
+                    }
+                }
+                if (current != null) {
+                    final MethodDescription.InDefinedShape selected = current;
+                    return new MethodVisitor(OpenedClassReader.ASM_API, methodVisitor) {
+                        @Override
+                        public void visitCode() {
+                            super.visitCode();
+                            Label label = new Label();
+                            super.visitLdcInsn(identifier);
```

Comment:
For these opcodes, could you add the pseudo Java code in a comment? Maybe even with the JVM bytecode in a comment as well. That would help with maintaining the overview, as reading the opcode instruction methods themselves can be quite difficult.

Reply:
Maybe at some point this could ship in bytebuddy.